jms/1.7/modules/ROOT/pages/jms-performance.adoc

```diff
@@ -1,47 +1,71 @@
+Connections are expensive to create, therefore reuse them as much as you can. By default, JMS Connector uses a caching strategy configuration that reuses as many consumers and producers as possible. Do not disable connection caching because modifying this configuration can reduce the performance of the application.
 
-== Don't Disable Connections Reuse
 
-Connections are expensive to create, so it's important to reuse them as much as possible.
+== Increase Concurrent Messages Processing
 
-*By default, the JMS Connector uses an aggressive* `caching-strategy` that reuses as much Consumers and Producers as possible. It is recommended to keep this configuration untouched, since any modification could reduce the performance of your application.
+To improve the performance of your application you can increase the number of concurrent consumers that receive messages from the same destination. JMS Connector enables you to increase the number of consumers by configuring the *Number of consumers* field in the *On New Message* source. By default, this source uses four consumers concurrently on the same destination, but you can increase this number to the one that fits your needs.
 
-WARNING: Disabling connections caching will reduce the performance of your application.
+In the following example, you increase the number of consumers:
 
+. In Studio, select the *On New Message* source from your flow.
+. In the *On New Message* source configuration screen, set the *Number of consumers* field to `20`.
 
-== Increasing Concurrent Messages Processing
+.JMS Connector Number of consumers Configuration
+image::jms-performance-1.png[In the On New Message configuration screen, set the Number of consumers field to 20]
 
-A good way of improving the performance of your application is to increase the number of concurrent consumers that receive messages from the same destination. This can be done easily in the JMS Connector by configuring the `numberOfConsumers` parameter.
-
-Out of the box, the `listener` will use *four consumers* concurrently on the same destination, but you can increase this number to the one that fits better your needs:
+In the XML editor, the `numberOfConsumers` configuration looks like this:
 
 [source,xml,linenums]
 ----
 <jms:listener config-ref="config" destination="#[vars.destination]" numberOfConsumers="20"/>
 ----
 
-TIP: Increasing the `numberOfConsumers` is the easiest way to improve the throughput of your listeners.
+== Optimize the Configuration for Clusters
+
+For applications running in clusters, keep in mind the concept of the primary node and how the connector behaves. When the app runs in a cluster, the *On New Message* source default behavior receives messages only in the primary node, no matter what kind of destination the app is consuming.
```

Comment:
"keep in mind" is idiomatic but common. Safe side:
```suggestion
For applications running in clusters, consider the primary node and how the connector behaves. When the app runs in a cluster, the *On New Message* source default behavior receives messages only in the primary node, no matter what kind of destination the app consumes.
```

Reply:
In some of the following edits, I assumed that the subject of "When consuming <blah>" is the connector. Please fix if not. The point is that the opening phrase "when doing this," must be followed by the thing doing it, to be grammatical (proper placement of modification).
For example, incorrect:
When eating quickly, _choking_ can occur.
Correct:
When eating quickly, _you_ might choke.
Does that make sense?